comp/core/autodiscovery/listeners/container.go

```diff
@@ -41,7 +41,7 @@ func NewContainerListener(options ServiceListernerDeps) (ServiceListener, error)
 	const name = "ad-containerlistener"
 	l := &ContainerListener{}
 	filter := workloadmeta.NewFilterBuilder().
-		SetSource(workloadmeta.SourceAll).
+		SetSource(workloadmeta.SourceRuntime).
```

Comment:
I don't remember the issue, But I think @L3n41c tried in the past to change this source but it has some side effect. 

the one that I'm thinking of (but not 100%sure) is when user don't give access to user to the container runtime, but we can still have info about the container with the kubelet.

Reply:
I see your point, but how about changing the source of the filter in the kubelet listener? 

Change was done [here](https://github.com/DataDog/datadog-agent/pull/22684/files#diff-8a283a9c39a018845b8b8c75721254c02380acf6798392c190c8737dedfe6706). Any idea why would we need to have events from all Sources?

Reply:
Keeping the container listener filter source as `SourceAll` and changing the kubelet's listener filter source back to `NodeOrchestrator` seems to also work (i.e. no duplicate logs, no extra warnings). 

I think also that since the `kubelet` is listening for events for `NodeOrchestrator`, then it also includes events from the kubelet about the containers. So it should still work IMO even if the container listener is using events from the container runtime and we don't have access to the runtime. 

WDYT?